src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ConstantExpectedAttribute.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    /// Indicates that the specified method parameter expects a constant
+    /// </summary>
+    /// <remarks>
+    /// This can be used to inform tooling that a constant should be used as an argument for the annotated parameter
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+    public sealed class ConstantExpectedAttribute : Attribute
+    {
+        /// <summary>
+        /// Indicates the minimum bounds of the constant expected
+        /// </summary>
+        public object? Min { get; set; }
+        /// <summary>
+        /// Indicates the maximum bounds of the constant expected
+        /// </summary>
+        public object? Max { get; set; }
```

Comment:
Nit: the XML comments should say whether the bound is inclusive or exclusive

Reply:
It wasn't mentioned in the issue but I'm assuming it's inclusive.